.github/workflows/test.yml

```diff
@@ -57,15 +57,25 @@ jobs:
             - name: setup Miniconda
               uses: conda-incubator/setup-miniconda@v2
 
-            - name: Setup Environment
-              run: make setup
+            - name: Create and activate virtual environment
+              run: |
+                  python3 -m venv test_venv
+                  source test_venv/bin/activate
+                  echo "venv: " $VIRTUAL_ENV
+
+            - name: Setup dependencies in virtual environment
+              run: |
+                  source test_venv/bin/activate
```

Comment:
If we are activating it here, we probably don't need to activate in the previous step?

Reply:
unless we really want to print details about it?

Reply:
Apparently, we do need to activate the `venv` in each named step. I tried it such that I activated it in an earlier step, but the later steps failed because the `venv` wasn't active. Googling around didn't produce anything meaningful to me that would fix this. @madhur-tandon - if you know the special incantation to make this work, please let me know. :+1:

Reply:
If you care to avoid it, you can do:
```sh
source test_venv/bin/activate
echo PATH=$PATH >> $GITHUB_ENV
echo VIRTUAL_ENV=$VIRTUAL_ENV >> $GITHUB_ENV
```
and then the environment will be active in all subsequent steps.

Reply:
So, I asked about not requiring the activation in the previous step because we don't really do anything with it apart from just printing information about it...

Reply:
the previous step being:

```
- name: Create and activate virtual environment
              run: |
                  python3 -m venv test_venv
                  source test_venv/bin/activate
                  echo "venv: " $VIRTUAL_ENV
```